lib/recipe/sync_to_local.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Recipe
+  class SyncToLocal
```

Comment:
This models the syncing of data from the API to a local json file as a side-effect of asking the API for data. So if you call `#all` on an instance of `SyncToLocal.new(LocalRecipes.new, RecipeApi.new)` it will save that data to the local file, but because it's modelled as a side-effect you won't know.
See for example https://github.com/thomcorley/recipe-finder/pull/4/files#diff-0d4ee945ec42eb115633dbb663bf1e5b0c2abd742695aa299d271933fa158390R13 where we do exactly that - request data from the API, but behind the scenes it's saving the data that was requested locally.

Reply:
Yeah, this is nice. When you call `#all` on `SyncToLocal.new(LocalRecipes.new, RecipeApi.new)` you're also oblivious to the fact that it's only downloading the recipes from the API when the recipe data is stale.

Reply:
I've been thinking about this a bit more and I think I'd maybe prefer to take this approach but not model it as a side-effect. Naming the class ApiClientWithLocalSyncing or something like that would make the code easier to read I think and it would be more obvious what is happening. What do you think? From my point of view it makes it easier to understand anyway!

Reply:
I'm going to cherry-pick from the ideas in this PR and incorporate them into my own branch 👌

Reply:
Actually, I've changed my mind and I like this "wrapper" class that separates the logic for syncing the data from the api to the local store. I will incorporate it into my main PR 👍